pkg/minikube/cluster/cluster.go

```diff
@@ -128,21 +121,53 @@ func DeleteHost(api libmachine.API) error {
 		return err
 	}
 
-	if err = unregister(host); err != nil {
-		return err
-	}
-
 	m := util.MultiError{}
 	m.Collect(host.Driver.Remove())
 	m.Collect(api.Remove(constants.MachineName))
 	return m.ToError()
 }
 
+// Register host VM
+func Register(api libmachine.API, instanceConfig *minishiftConfig.InstanceConfigType) error {
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
+		return err
+	}
+	if !instanceConfig.IsRegister {
+		registered, err := registration.RegisterHostVM(host, RegistrationParameters)
+		if err != nil {
+			return err
+		}
+		if registered {
+			instanceConfig.IsRegister = true
+			if err := instanceConfig.Write(); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
 // Un-register host VM
-func unregister(host *host.Host) error {
+func UnRegister(api libmachine.API, instanceConfig *minishiftConfig.InstanceConfigType) error {
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
+		return err
+	}
+
+	if !instanceConfig.IsRegister {
+		return nil
+	}
 	if drivers.MachineInState(host.Driver, state.Running)() {
-		if err := registration.UnregisterHostVM(host, RegistrationParameters); err != nil {
-			return fmt.Errorf("Error unregistring the VM: %s", err)
+		unRegistered, err := registration.UnregisterHostVM(host, RegistrationParameters)
+		if err != nil {
+			return err
+		}
+		if unRegistered {
```

Comment:
See https://github.com/minishift/minishift/pull/988/files#r120873570. I don't think  UnRegister needs to know anything off InstanceConfigType.